ads/google/a4a/traffic-experiments.js

```diff
@@ -184,21 +186,27 @@ export function googleAdsIsA4AEnabled(win, element, experimentName,
  *   parameter or not.
  */
 function maybeSetExperimentFromUrl(win, element, experimentName,
-    controlBranchId, treatmentBranchId, delayedControlId,
-    delayedTreatmentBrandId, sfgControlId, sfgTreatmentId, manualId) {
+     controlBranchId, treatmentBranchId, delayedControlId,
+     delayedTreatmentBrandId, sfgControlId, sfgTreatmentId, manualId) {
   const expParam = viewerForDoc(element).getParam('exp') ||
-      parseQueryString(win.location.search)['exp'];
+    parseQueryString(win.location.search)['exp'];
   if (!expParam) {
     return false;
   }
-  const match = /(^|,)(a4a:[^,]*)/.exec(expParam);
-  const a4aParam = match && match[2];
-  if (!a4aParam) {
+  // Allow for per type experiment control with Doubleclick key set for 'da'
+  // and AdSense using 'aa'.  Fallbsck to 'a4a' if type specific is missing.
+  const expKeys = [
+    (element.getAttribute('type') || '').toLowerCase() == 'doubleclick' ?
+      'da' : 'aa',
+    'a4a',
+  ];
+  let arg;
+  let match;
+  expKeys.forEach(key => arg = arg ||
+    (match = new RegExp(`(?:^|,)${key}:(-?\\d+)`).exec(expParam)) && match[1]);
```

Comment:
Match is assigned as the second part of the predicate and if were undefined, match[1] would not execute so I'm not sure what the concern is.